Miscellaneous/gdpr/pseudonymization/README.md

```diff
@@ -0,0 +1,177 @@
+
+
+#### Fingerprint Filter Approach
+
+4. To utilise the first approach, based on the FingerPrint Filter execute the following command:
+
+    `cat sample_docs | nc localhost 5000`
+    
+    This should also take a few seconds to execute and index the 100 documents from the sample file.
+
+#### Ruby Script File Approach
+    
+5. To utilise the second approach, based on the Ruby File Script execute the following command:
+
+    `cat sample_docs | nc localhost 6000`
+   
+   This should also take a few seconds to execute and index the 100 documents from the sample file.
+
+#### Inspecting and using the data
+
+6. Pseudonymized Documents will be indexed to an `events` index.  These can be accessed through the following query:
+
+    `curl "http://localhost:9200/events/_search" -u elastic:changeme | jq`
+    
+    Example pseudonymized document below:
+    
+    ```shell
+        {
+            "_index": "events",
+            "_type": "doc",
+            "_id": "tQOjQ2IBED8Jv9YVVDxs",
+            "_score": 1,
+            "_source": {
+              "host": "gateway",
+              "user_agent": "Mozilla/5.0 (Macintosh; PPC Mac OS X 10_6_7) AppleWebKit/535.1 (KHTML, like Gecko) Chrome/14.0.790.0 Safari/535.1",
+              "job_title": "Electrical Engineer",
+              "username": "95b88d8d477e18a8acca833e7bcbd2c5d5f646b29e2d1c9604a1d930e2f63313",
+              "@timestamp": "2018-03-20T13:39:59.799Z",
+              "ip": "e85022a9801b356dd8c3ed6b2e02f0061a3aeea5bbad15a9ff4aed35b5bb3a42",
+              "source": "ruby_pipeline",
+              "city": "Komsomol’skiy",
+              "title": "Mr",
+              "country_code": "UZ",
+              "@version": "1",
+              "gender": "Female",
+              "country": "Uzbekistan",
+              "port": 41126
+            }
+         }
+    ```
+    
+    "Identity documents" (effectively key-value pair lookups), are indexed into a `identities` index.  These can be accessed through the following query:
+    
+    `curl "http://localhost:9200/identities/_search" -u elastic:changeme | jq`
+    
+    Again we assume jq is available for display of results.
+    
+    ```shell
+      {
+        "_index": "identities",
+        "_type": "doc",
+        "_id": "1924d02bd98a46c795cb2a925b98a22ae59c563e0de49f4ba4aa49e6cab072ad",
+        "_score": 1,
+        "_source": {
+          "key": "1924d02bd98a46c795cb2a925b98a22ae59c563e0de49f4ba4aa49e6cab072ad",
+          "value": "174.145.248.21",
+          "tags": [
+            "identities"
+          ],
+          "@timestamp": "2018-03-20T13:39:59.957Z",
+          "@version": "1",
+          "source": "ruby_pipeline"
+        }
+      } 
+      ```
+
+#### Data Notes
+
+
+   * The data produced by both examples is identical. If running both examples once, you will end up with a duplicate of each document in the `events` index - total 200, and 100 thereafter for each execution.
```

Comment:
remove this line, you mention the same thing again two lines down

Reply:
I don't see the repetition im refering to different indices.

Reply:
ah sorry see thanks